File: vital-commands-processor/src/main/java/me/xra1ny/vital/commands/processor/VitalCommandsProcessorSubModule.java
package me.xra1ny.vital.commands.processor;

import me.xra1ny.essentia.inject.annotation.Component;
import me.xra1ny.vital.VitalSubModule;
import org.jetbrains.annotations.NotNull;

/**
 * Defines the vital-commands-processor submodule.
 */
@Component
public class VitalCommandsProcessorSubModule extends VitalSubModule {
    @Override
    @NotNull
    public String getName() {
        return "vital-commands-processor";
    }
}

File: vital-commands-processor/src/main/kotlin/me/vitalframework/commands/processor/VitalCommandsProcessorSubModule.kt
package me.vitalframework.commands.processor

import me.vitalframework.SubModule
import me.vitalframework.VitalCoreSubModule.Companion.logger
import me.vitalframework.VitalSubModule

/**
 * Defines the official vital-commands-processor submodule, which is displayed when Vital starts.
 * This submodule should never be used in the final plugin jar.
 * It should only exist during project compilation.
 */
@SubModule("vital-commands-processor")
class VitalCommandsProcessorSubModule : VitalSubModule() {
    val logger = logger()

    override fun onInstall() {
        logger.error("'vital-commands-processor' has been installed but this submodule should not be included in your plugin.")
        logger.error("You should exclude 'vital-commands-processor' from your build as it is only relevant during compilation.")
    }
}
